views/includes/global-header-next.njk

```diff
@@ -103,8 +145,65 @@
             </div>
 
             <div class="global-header-next__search">
-                {{ navSearchForm() }}
+                {{ globalSearch() }}
             </div>
         </div>
     </div>
 </header>
+
+<script>
+/**
+ * Initialise event listeners on header navigation early
+ * (e.g. search and menu toggle)
+ *
+ * We do this early so that users can interact
+ * with the navigation before the main script bundle has loaded
+ * increasing time-to-interactive
+ */
+(function() {
+    const elSelector = '.js-global-header';
+    var stateClassNames = {
+        nav: 'has-toggled-navigation',
+        search: 'has-toggled-search'
+    };
+
+    var body = document.querySelector('body');
+    var html = document.querySelector('html');
+    var el = document.querySelector(elSelector);
+    var searchInput = el.querySelector('input[type=search]');
+
+    if (!el) {
```

Comment:
You could maybe move this up a bit earlier and bail out before doing the other queries?